cmp/options.go

```diff
@@ -35,79 +38,33 @@ type Option interface {
 // on all individual options held within.
 type Options []Option
 
-func (Options) option() {}
-
-type (
-	pathFilter  func(Path) bool
-	valueFilter struct {
-		in  reflect.Type  // T
-		fnc reflect.Value // func(T, T) bool
+func (opts Options) filter(s *state, vx, vy reflect.Value, t reflect.Type) (out Option) {
+	for _, opt := range opts {
+		switch opt := opt.filter(s, vx, vy, t); opt.(type) {
+		case ignore:
+			return ignore{} // Only ignore can short-circuit evaluation
+		case invalid:
+			out = invalid{} // Takes precedence over comparer or transformer
+		case *comparer, *transformer:
+			switch out.(type) {
+			case nil:
+				out = opt
+			case invalid:
+				// Keep invalid
+			case *comparer, *transformer:
```

Comment:
This is incorrect. I have to batch all *comparers and *transformers since it is still possible that a later `Ignore` will take precedence over all these conflicts.